crates/bitwarden/src/platform/client_fido.rs

```diff
@@ -0,0 +1,47 @@
+use std::sync::{Arc, Mutex};
+
+use bitwarden_fido::{
+    Fido2Authenticator, Fido2Client, Fido2CredentialStore, Fido2UserInterface,
+    FidoEncryptionSettingStore,
+};
+
+use crate::Client;
+
+pub struct ClientFido2<'a> {
+    #[allow(dead_code)]
+    pub(crate) client: &'a Client,
+}
+
+impl FidoEncryptionSettingStore for Client {
+    fn get_encryption_settings(
+        &self,
+    ) -> Result<Arc<dyn bitwarden_crypto::KeyContainer>, bitwarden_core::VaultLocked> {
+        Ok(self.get_encryption_settings()?)
+    }
+}
+
+impl<'a> ClientFido2<'a> {
+    pub fn create_authenticator(
+        &'a self,
+        user_interface: &'a dyn Fido2UserInterface,
+        credential_store: &'a dyn Fido2CredentialStore,
+    ) -> Fido2Authenticator<'a> {
+        Fido2Authenticator {
+            client: self.client,
+            user_interface,
+            credential_store,
+            selected_cipher: Mutex::new(None),
```

Comment:
Could we create a `Fido2Authenticator::new()` that doesn't expose `selected_cipher` and `requested_uv`, and make those two fields `pub(crate)`? They feel like implementation details that shouldn't be exposed.